crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -89,7 +89,11 @@ enum ResolverVersion {
     Unavailable(Version, UnavailableVersion),
 }
 
-pub struct Resolver<'a, Provider: ResolverProvider> {
+pub struct Resolver<
+    'a,
+    Provider: ResolverProvider,
+    InstalledPackages: InstalledPackagesProvider + Send + Sync,
```

Comment:
I wonder about the value of making this generic. It means we need separate implementations for each concrete type when compiling. I know it's used for the empty package, but could that just be represented by an empty `SitePackages`?

Reply:
I think it needs to be generic because of the crate dependencies, I didn't do it to have an empty one. If it's not generic, then `uv-resolver` has a dependency on `uv-installer` which afaik is a no-no and the purpose of the `uv-traits` crate.

Reply:
Ahh okay, I see. That makes sense.